src/coreclr-debug/debugConfigurationProvider.ts

```diff
@@ -59,6 +61,47 @@ export class DotnetDebugConfigurationProvider implements vscode.DebugConfigurati
                 return undefined;
             }
         }
+        
+        // We want to ask the user if we should run dotnet  dev-certs https --trust, but this doesn't work in a few cases --
+        // Linux -- not supported by the .NET CLI as there isn't a single root cert store
+        // VS Code remoting/Web UI -- the trusted cert work would need to happen on the client machine, but we don't have a way to run code there currently
+        // pipeTransport -- the dev cert on the server will be different from the client
+        if (!this.platformInformation.isLinux() && !vscode.env.remoteName && vscode.env.uiKind != vscode.UIKind.Web && !debugConfiguration.pipeTransport)
+        {
+            if(debugConfiguration.checkForDevCert === undefined && debugConfiguration.serverReadyAction)
+            {
+                debugConfiguration.checkForDevCert = true;
+            }
+
+            if (debugConfiguration.checkForDevCert)
+            {
+                hasDotnetDevCertsHttps(this.options.dotNetCliPaths).then(async (returnData) => {
+                    if(returnData.error) //if the prcess returns 0 error is null, otherwise the return code can ba acessed in returnData.error.code
+                    {
+                        const result = await vscode.window.showInformationMessage(
+                            "The selected launch configuration is configured to launch a web browser but no trusted development certificate was found. Create a trusted self-signed certificate?", 
+                            { title:"Yes"}, { title:'Not Now', isCloseAffordance: true }, { title:"Don't Ask Again"}
+                            ); 
+                        if (result?.title === 'Yes')
+                        {
+                            let returnData = await createSelfSignedCert(this.options.dotNetCliPaths);
+                            if (!returnData.error)
+                            {
+                                vscode.window.showInformationMessage('Self-signed certificate sucessfully created');
+                            }
+                            else
+                            {
+                                vscode.window.showWarningMessage("Couldn't create self-signed certificate");
```

Comment:
>") [](http://example.com/codeflow?start=105&length=2)

You want to do something with the failure stuff you got. Probably, to write it to the C# log, and updated the message to be something like "Couldn't create self-signed certificate. See <something> for more information."